src/pipx/main.py

```diff
@@ -614,19 +616,68 @@ def get_command_parser() -> argparse.ArgumentParser:
     return parser
 
 
+def setup_file_handler() -> logging.Handler:
```

Comment:
I have not found the built in python logging utilities to be too user friendly, but I wonder if there is something built in to it that automatically rotates files and takes care of the dirty work for you. Here are a couple things I found.

https://www.blog.pythonlibrary.org/2014/02/11/python-how-to-create-rotating-logs/
https://docs.python.org/3/library/logging.handlers.html#baserotatinghandler
https://docs.python.org/3/library/logging.handlers.html#logging.handlers.RotatingFileHandler

Reply:
I mentioned that in the blurb for this PR. I started using `RotatingFileHandler`.  It's not a perfect fit.  It wants to keep appending to the current file until it gets too big at which point it rolls all the files.  I wanted a separate log file for every command invocation.

I managed to used it anyway in the beginning, by manualling initiating a file roll with `doRollover()`, but this caused a crash on Windows, because by creating the handler it opened the log file, and then manually trying to rename that open file made Windows upset.

Reply:
Whoops sorry about that. I see it now. Custom implementation sounds good to me!